Makefile.include

```diff
@@ -705,6 +705,9 @@ CFLAGS += -include '$(RIOTBUILD_CONFIG_HEADER_C)'
 # include mcuboot support
 include $(RIOTMAKE)/mcuboot.mk
 
+# include bootloaders support
+include $(RIOTMAKE)/boot/riotboot.mk
```

Comment:
This include should be before the `link` target definition.

Because when trying to overwrite one variable like `HEXFILE`, the value will already have been evaluated when declaring the `link` target.
And later, selecting the right file to flash should only be `FLASHFILE = $(NAMED_OF_THE_COMBINED_FILE)` before including `Makefile.include`.

In general the rules should be the last things defined even if in `Makefile.include` is not completely the case for now.